services/fonts.js

```diff
@@ -0,0 +1,223 @@
+/**
+ * Add a font
+ *
+ * See [the corresponding HTTP service documentation](https://docs.mapbox.com/api/maps/fonts/#add-a-font).
+ *
+ * @param {Object} config
+ * @param {UploadableFile} config.file - A TTF or OTF file.
+ * @param {string} [config.ownerId]
+ * @return {MapiRequest}
+ *
+ * @example
+ * fontsClient.createFont({
+ *   file: 'path/to/file.ttf'
+ * })
+ *   .send()
+ *   .then(response => {
+ *     const newFont = response.body;
+ *   });
+ */
+Fonts.createFont = function(config) {
+  v.assertShape({
+    file: v.file,
+    ownerId: v.string
+  })(config);
+
+  return this.client.createRequest({
+    method: 'POST',
+    path: '/fonts/v1/:ownerId',
+    params: pick(config, ['ownerId']),
+    file: config.file
+  });
+};
+
+/**
+ * Delete a font.
+ *
+ * See [the corresponding HTTP service documentation](https://docs.mapbox.com/api/maps/fonts/#delete-a-font).
+ *
+ * @param {Object} config
+ * @param {string} config.font
+ * @param {string} [config.ownerId]
+ * @return {MapiRequest}
+ *
+ * @example
+ * fontsClient.deleteFont({
+ *   font: 'Custom Font Regular'
+ * })
+ *   .send()
+ *   .then(response => {
+ *     // delete successful
+ *   });
+ */
+Fonts.deleteFont = function(config) {
+  v.assertShape({
+    font: v.required(v.string),
+    ownerId: v.string
+  })(config);
+
+  return this.client.createRequest({
+    method: 'DELETE',
+    path: '/fonts/v1/:ownerId/:font',
+    params: config
+  });
+};
+
+/**
+ * Get font metadata.
+ *
+ * See the [corresponding HTTP service documentation](https://docs.mapbox.com/api/maps/fonts/#retrieve-font-metadata).
+ *
+ * @param {Object} config
+ * @param {string} config.font
+ * @param {string} [config.ownerId]
+ * @return {MapiRequest}
+ *
+ * @example
+ * fontsClient.getFontMetadata({
+ *   font: 'Custom Font Regular'
+ * })
+ *   .send()
+ *   .then(response => {
+ *     const metadata = response.body;
+ *   });
+ */
+Fonts.getFontMetadata = function(config) {
+  v.assertShape({
+    font: v.required(v.string),
+    ownerId: v.string
+  })(config);
+
+  return this.client.createRequest({
+    method: 'GET',
+    path: '/fonts/v1/:ownerId/:font/metadata',
+    params: pick(config, ['ownerId', 'font'])
+  });
+};
+
+/**
+ * Update font metadata.
+ *
+ * See the [corresponding HTTP service documentation](https://docs.mapbox.com/api/maps/fonts/#update-font-metadata).
+ *
+ * @param {Object} config
+ * @param {string} config.font
+ * @param {String} config.visibility - `visibility` property of font metadata. The only valid values are `public` and `private`.
+ * @param {string} [config.ownerId]
+ * @return {MapiRequest}
+ *
+ * @example
+ * fontsClient.updateFontMetadata({
+ *   font: 'Custom Font Regular',
+ *   visibility: 'public'
+ * })
+ *   .send()
+ *   .then(response => {
+ *     const metadata = response.body;
+ *   });
+ */
+Fonts.updateFontMetadata = function(config) {
+  v.assertShape({
+    font: v.required(v.string),
+    visibility: v.required.oneOf(['public', 'private']),
```

Comment:
Ah, sorry I got the syntax wrong in my review suggestion, @andrewharvey, which is why tests are failing. Should be:

```js
v.required(v.oneOf(['public', 'private']))
```